application/forms/EventRuleConfigForm.php

```diff
@@ -415,62 +463,50 @@ protected function createRemoveButton(string $prefix): SubmitButtonElement
      *
      * @return int
      */
-    public function addOrUpdateRule(int $id, array $config): int
+    public function updateRule(int $id, array $config): int
     {
         $db = Database::get();
-
         $db->beginTransaction();
+        $db->update('rule', ['object_filter' => $this->objectFilter], ['id = ?' => $id]);
```

Comment:
Please only update `object_filter` if it has been changed.

Reply:
If there are no changes, the transaction does not even happen.

Reply:
As discussed offline, this is also triggered if we only add a new escalation and the filter remains the same.